### PR TITLE
PT-850: Pending in Plugins for SW6

### PR DESCRIPTION
**Pull request by ivan-pugach-mondu**
## Description

This PR aims to improve the order process and assign the proper status on the Shopware side when the payment goes through the pending flow

Release: p
Tickets: PT-850

Fixes # (issue)
https://mondu.atlassian.net/browse/PT-850

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Yaroslavzev — DISMISSED** [review]
utACK 👍 

**suggestion(non-blocking)**: from my prospective it would be nice to have place with conditions and it will help to maintain codebase in the future. This place can be hash, service object which contain mapping between transaction and responce statuses or humane readable method. 
For example this two lines
```
                $confirmResponseState != 'confirmed' &&
                $confirmResponseState != 'pending'
```
can be moved to method with name `orderConfirmed?`
Please note that I am fine to leave as it's now.

File: src/Components/Webhooks/Controller/WebhooksController.php [hunk 1]
```diff
@@ -52,7 +52,6 @@ public function process(Request $request, Context $context): Response
         case 'order/pending':
             [$resBody, $resStatus] = $this->webhookService->handlePending($params, $context);
             break;
-        case 'order/canceled':
```
Comment (arthurmmoreira) [inline src/Components/Webhooks/Controller/WebhooksController.php:55]: Nice 👍

**Review by arthurmmoreira — APPROVED** [review]
utACK

**Comment by ume3119** [comment]
@ivan-pugach-mondu @tikohov20  can it be deployed on our stage for testing?

**Comment by tikohov20** [comment]
> @ivan-pugach-mondu @tikohov20 can it be deployed on our stage for testing?

I'll deploy it to stage.

File: src/Components/PaymentMethod/PaymentHandler/MonduHandler.php [hunk 2]
```diff
@@ -269,4 +272,9 @@ public function createLocalOrder($transaction, $orderUuid, $salesChannelContext)
             ]
         ], $salesChannelContext->getContext());
     }
+
+    protected function isOrderConfirmed($confirmResponseState)
+    {
+        return $confirmResponseState != self::RESPONSE_STATE_CONFIRMED && $confirmResponseState != self::RESPONSE_STATE_PENDING;
```
Comment (tikohov20) [inline src/Components/PaymentMethod/PaymentHandler/MonduHandler.php:278]: is this part correct?

for example lets assume $confirmResponseState is confirmed

in this case $confirmResponseState != confirmed so this function will return false.

I think we should change this to 

`
return in_array($confirmResponseState, [self::RESPONSE_STATE_CONFIRMED, self::RESPONSE_STATE_PENDING]);
`

**Review by tikohov20 — CHANGES_REQUESTED** [review]

**Comment by tikohov20** [comment]
Lets update the composer.json file and update the version there in this pr.

File: src/Components/PaymentMethod/PaymentHandler/MonduHandler.php (hunk 2 above)
Comment (ivan-pugach-mondu) [reply src/Components/PaymentMethod/PaymentHandler/MonduHandler.php]: Thank you @tikohov20, good catch. I did the update, please review once again

**Review by tikohov20 — DISMISSED** [review]

**Review by tikohov20 — APPROVED** [review]

**Pull request merged**